ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -265,6 +231,40 @@ public Mode suspendScan (boolean async)
    {
        synchronized (_stateLock)
        {
+           // see if suspension should delay until certain types have recovered
```

Comment:
@jmfinelli do you think this whole block should be guarded by the async check?

Reply:
Nope, I don't think so. `async` should not interfere with this logic since it checks for left-over transactions to wait upon. `async` should be used only to wait for any in-progress scan to complete before returning.

TBH, I think we are forcing what `suspendScan` was originally designed for. In a way, we are "overloading" `suspendScan` with a global switch (i.e. `recoveryPropertyManager.getRecoveryEnvironmentBean().isWaitForFinalRecovery()`). This feels far from ideal. Personally, I would stick with Jonathan's [comment](https://github.com/jmfinelli/personal-narayana-proposals/pull/1#issuecomment-1240702395).

Reply:
I guess the other option would be to provide an additional parameter to the suspendScan function. That being said, I suppose the async applicable to every call to suspendScan but this "return from suspend when conditions are met" is meant to be.

Reply:
I haven't addresses this but did make other changes, please can you take a look at the current branch state https://github.com/tomjenkinson/narayana/compare/a01440a52e8b9165b44ccc48d99955a71f7e86b9...26cab2bd194617e264f4cc666e85b3c32f522494 ?